### DISCUSSION TITLE
nuxt3 pinia  localStorage is not defined

### DISCUSSION THREAD

**Discussion by fxzer** [vuejs/pinia#2336] [Help and Questions] [answered]
### Reproduction

nuxt3 pinia  localStorage is not defined

### Steps to reproduce the bug

![image](https://github.com/vuejs/pinia/assets/68853957/6c4ee03f-fb98-4ee1-bef9-e48002482d6e)
![image](https://github.com/vuejs/pinia/assets/68853957/f60fc836-c4ef-4397-ab7f-5c4128b8ee37)

### Expected behavior

can use localStorage 

### Actual behavior

can use localStorage

**Comment by posva** [comment, marked as answer]
That error happens on server because localStorage is browser only (not related to pinia)
I recommend you to use or check Vue use useLocalStoarge to handle things

**Comment by santiagodeheza** [comment]
Quick update, if it helps anyone, to avoid this error from happening while using vite-ssg, you can use `isClient`, i.e

```js
// This plugin is used for state persistence in Pinia stores.
import piniaPluginPersistedstate from 'pinia-plugin-persistedstate'

export const install = ({ isClient, app, router }) => {
  // Create a new Pinia instance.
  const pinia = createPinia()

  pinia.use(({ store }) => {
    // The 'router' instance is marked as raw using 'markRaw'.
    // This prevents Vue from observing and tracking it in the reactivity system.
    store.router = markRaw(router)
  })

  if (isClient) {
    //Very important to only do this client-side, to avoid localStorage being undefined during ssg-build.
    // Use the 'piniaPluginPersistedstate' for state persistence in Pinia stores.
    pinia.use(piniaPluginPersistedstate)
  }
  app.use(pinia)
}

```

otherwise, you will get console errors like `ReferenceError: localStorage is not defined`
